wemake_python_styleguide/visitors/tokenize/comments.py

```diff
@@ -51,6 +52,16 @@ def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         self._no_cover_count = 0
 
+        # Variables for WPS#463 EmptyCommentViolation
+        self._prev_comment_line_num = -1
+        self._prev_non_empty = -1
+        self._in_same_block = True
+        self._block_alerted = False
+        self._reserved_token = tokenize.TokenInfo(
```

Comment:
Can you please explain this variable?

Reply:
Our approach is parsing a block of comment at once since there should only be at most one error alerted per block. The only time``self._in_same_block`` is false is when we are reading a new block, i.e. the first line of a comment block. (If it's empty, we know to alert immediately) ``self._block_alerted`` is used to prevent more than one alert within the same block.

``self._prev_non_empty`` is used to keep track of the line number of previous non-empty comment. It is mostly for this kind of situations: 
```python
# Non-empty 1
# Non-empty 2
#
#
# We are not here yet, the existence of this line affects whether we alert on line 3 or not.
```
When reading line 3, the token for line 3 will be assigned to ``self._reserved_token`` for a potential alert. And line 4 will be ignored because it is not right after a non-empty comment. (We did ``current line number - self._prev_non_empty`` == 1 to check). It is "reserved" because we should only alert if there is no non-empty comment in the same block later.